### PR TITLE
introduce a new mod `Illusionist`

### PR DESCRIPTION
**Pull request by boscohyun**
SSIA

NineChronicles Mods Illusionist-PoC-01

### CODE REVIEW

File: NineChronicles.Mods.Illusionist/Tools/TextureLoader.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            var pixels = tex.GetPixels(); // get pixel colors
+            for (var i = 0; i < pixels.Length; i++)
+            {
+                pixels[i].a = pixels[i].grayscale; // set the alpha of each pixel to the grayscale value
+            }
+
+            tex.SetPixels(pixels); // set changed pixel alphas
+            tex.Apply(); // upload texture to GPU
+            return tex;
+        }
```
Comment (moreal) [inline NineChronicles.Mods.Illusionist/Tools/TextureLoader.cs:58]: It seems unused

File: NineChronicles.Mods.Illusionist/Tools/TextureLoader.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            Graphics.Blit(source, renderTex);
+            RenderTexture previous = RenderTexture.active;
+            RenderTexture.active = renderTex;
+            Texture2D readableText = new Texture2D(source.width, source.height);
+            readableText.ReadPixels(new Rect(0, 0, renderTex.width, renderTex.height), 0, 0);
+            readableText.Apply();
+            RenderTexture.active = previous;
+            RenderTexture.ReleaseTemporary(renderTex);
+            return readableText;
+        }
```
Comment (moreal) [inline NineChronicles.Mods.Illusionist/Tools/TextureLoader.cs:78]: It seems unused

File: NineChronicles.Mods.Illusionist/NineChronicles.Mods.Illusionist.csproj [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        </Reference>
+    </ItemGroup>
+
+    <ItemGroup>
+        <Reference Include="Nekoyume">
+            <HintPath>$(NINECHRONICLES_MANAGED_DIR)\Nekoyume.dll</HintPath>
+        </Reference>
+<!--        <Reference Include="UniTask">-->
+<!--            <HintPath>$(NINECHRONICLES_MANAGED_DIR)\UniTask.dll</HintPath>-->
+<!--        </Reference>-->
```
Comment (moreal) [inline NineChronicles.Mods.Illusionist/NineChronicles.Mods.Illusionist.csproj:34]:
```suggestion
        </Reference>
```

File: NineChronicles.Mods.Illusionist/Manifest.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+namespace NineChronicles.Mods.Illusionist
+{
+    public static class Manifest
+    {
+        public const string PluginGuid = "org.ninechronicles.mods.illusionist";
```
[the remark this answers is not in the document]
Comment (boscohyun) [reply NineChronicles.Mods.Illusionist/Manifest.cs:5]: It is an argument name of `BepInPlugin` attribute.

```cs
namespace NineChronicles.Mods.Illusionist
{
    [BepInPlugin(Manifest.PluginGuid, Manifest.PluginName, Manifest.PluginVersion)]
    public class IllusionistPlugin : BaseUnityPlugin
    {
        // ...
    }

    // ...
}

namespace BepInEx
{
  /// <summary>
  /// This attribute denotes that a class is a plugin, and specifies the required metadata.
  /// </summary>
  [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
  public class BepInPlugin : Attribute
  {
    /// <summary>
    /// The unique identifier of the plugin. Should not change between plugin versions.
    /// </summary>
    public string GUID { get; protected set; }

    /// <summary>
    /// The user friendly name of the plugin. Is able to be changed between versions.
    /// </summary>
    public string Name { get; protected set; }

    /// <summary>The specfic version of the plugin.</summary>
    public Version Version { get; protected set; }

    /// <param name="GUID">The unique identifier of the plugin. Should not change between plugin versions.</param>
    /// <param name="Name">The user friendly name of the plugin. Is able to be changed between versions.</param>
    /// <param name="Version">The specfic version of the plugin.</param>
    public BepInPlugin(string GUID, string Name, string Version)
    {
      this.GUID = GUID;
      this.Name = Name;
      try
      {
        this.Version = new Version(Version);
      }
      catch
      {
        this.Version = (Version) null;
      }
    }

    // ...
}
```

**Review by moreal — APPROVED** [review]

**Pull request merged**